kbase/kbase-ui
# UIP-11: auth2-client xss health

# Pull Request

## Description

This updates the auth2-client plugin with closing an xss exposure or two, and more generally wraps all calls to innerHTML with a wrapper function which applies sanitization via DOMPurify. This is overkill, yes, but does not rely on minutely verifying all data injected into html markup.

In addition, a few small fixes were applied, primarily in the token logout features.

Manually tested all auth workflows.

## Issues Resolved

https://kbase-jira.atlassian.net/browse/UIP-11

* [x] Added the Jira Tickets to the title of the PR e.g. (PTV-XXX fixes a terrible bug)
* [-] Added the Github Issue to the title of the PR e.g. (PTV-XXX adds an awesome feature)

## Testing Instructions

The integration tests generally verify that the plugin loads correctly and does a few tasks, but they should be expanded greatly to cover more use cases. Full coverage may be difficult, e.g. signin and signup, account linking
  
* [x] Tests pass in github actions
* [x] Manually verified that changes area available (by spinning up an instance and navigating to _X_ to see _Y_)

## Dev Checklist

* [-] Code follows the guidelines at [https://sites.google.com/truss.works/kbasetruss/development](https://sites.google.com/truss.works/kbasetruss/development)
* [x] Integration tests have been run and fully pass (only when preparing a release)
* [x] I have run run the code quality script against the codebase (also done implicitly during a build)

## Release Notes

* [x] Ensure there is an "upcoming release notes" file located in release-notes/RELEASE_NOTES_NEXT.md
* [x] Add relevant notes to this document
